streams/src/test/java/org/apache/kafka/streams/integration/SuppressionDurabilityIntegrationTest.java

```diff
@@ -0,0 +1,275 @@
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.integration.utils.EmbeddedKafkaCluster;
+import org.apache.kafka.streams.integration.utils.IntegrationTestUtils;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.kstream.Serialized;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.lang.Long.MAX_VALUE;
+import static java.time.Duration.ofMillis;
+import static java.util.Arrays.asList;
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.apache.kafka.common.utils.Utils.mkProperties;
+import static org.apache.kafka.streams.StreamsConfig.AT_LEAST_ONCE;
+import static org.apache.kafka.streams.StreamsConfig.EXACTLY_ONCE;
+import static org.apache.kafka.streams.kstream.Suppressed.BufferConfig.maxRecords;
+import static org.apache.kafka.streams.kstream.Suppressed.untilTimeLimit;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(Parameterized.class)
+@Category({IntegrationTest.class})
+public class SuppressionDurabilityIntegrationTest {
+    @ClassRule
+    public static final EmbeddedKafkaCluster CLUSTER = new EmbeddedKafkaCluster(3);
+    private static final StringDeserializer STRING_DESERIALIZER = new StringDeserializer();
+    private static final StringSerializer STRING_SERIALIZER = new StringSerializer();
+    private static final Serde<String> STRING_SERDE = Serdes.String();
+    private static final LongDeserializer LONG_DESERIALIZER = new LongDeserializer();
+    private static final int COMMIT_INTERVAL = 100;
+    private static final int SCALE_FACTOR = COMMIT_INTERVAL * 2;
+    private static final long TIMEOUT_MS = 30_000L;
+    private final boolean eosEnabled;
+
+    public SuppressionDurabilityIntegrationTest(final boolean eosEnabled) {
+        this.eosEnabled = eosEnabled;
+    }
+
+    @Parameters(name = "{index}: eosEnabled={0}")
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(new Object[] {false}, new Object[] {true});
+    }
+
+    private KTable<String, Long> buildCountsTable(final String input, final StreamsBuilder builder) {
```

Comment:
this seems to be 100% code duplication with the other test. Maybe worth move this so an helper (or abstract class) and unify? (similar other helper methods below)

Reply:
This really doesn't seem right to me.

I've already unified as much as I felt appropriate by extracting several methods into IntegrationTestUtils, which is the place where we already share common code.

Beyond that, though, these are two different tests. They only do the same topology coincidentally. It seems needlessly complex to introduce an abstract class or some other secondary "test util" class to de-duplicate a few blocks of trivially simple code... It just makes the tests harder to read and maintain.

Reply:
Since this particular method has just one usage in this file, I will go ahead and inline it, though.

Reply:
I've pulled more of the common utility code into IntegrationTestUtils. Does it still seem too duplicate-y?

Reply:
It's ok :) You reasoning about "same topology by change" makes sense.